sdk/cosmos/src/operations/create_database.rs

```diff
@@ -3,44 +3,71 @@ use crate::prelude::*;
 use crate::resources::Database;
 use crate::ResourceQuota;
 use azure_core::headers::{etag_from_headers, session_token_from_headers};
-use azure_core::{collect_pinned_stream, Request as HttpRequest, Response as HttpResponse};
+use azure_core::{collect_pinned_stream, Context, PipelineContext, Response as HttpResponse};
 use chrono::{DateTime, Utc};
 
 #[derive(Debug, Clone)]
-pub struct CreateDatabaseOptions {
+pub struct CreateDatabaseBuilder {
+    client: CosmosClient,
+    database_name: String,
     consistency_level: Option<ConsistencyLevel>,
+    context: Context,
 }
 
-impl CreateDatabaseOptions {
-    pub fn new() -> Self {
+impl CreateDatabaseBuilder {
+    pub(crate) fn new(client: CosmosClient, database_name: String) -> Self {
         Self {
+            client,
+            database_name,
             consistency_level: None,
+            context: Context::new(),
         }
     }
 
     setters! {
         consistency_level: ConsistencyLevel => Some(consistency_level),
+        context: Context => context,
     }
-}
 
-impl CreateDatabaseOptions {
-    pub(crate) fn decorate_request(
-        &self,
-        request: &mut HttpRequest,
-        database_name: &str,
-    ) -> crate::Result<()> {
-        #[derive(Serialize)]
-        struct CreateDatabaseRequest<'a> {
-            pub id: &'a str,
-        }
-        let req = CreateDatabaseRequest { id: database_name };
+    pub fn insert<E: Send + Sync + 'static>(&mut self, entity: E) -> &mut Self {
```

Comment:
> What about having a trait for this function? It won't change much but it would signal unequivocally the meaning of this.

We discussed this, and adding a trait means that users of the method would need to import it into scope before being able to use it. That's not necessarily a bad idea, but before we decide to do that we should probably understand better how often the context interface is used.

We settled on keeping it as an inherent function for this proposal, but leaving the option open to re-evaluate once we have a better grip on usage.